test/changeset_upload_observation.js

```diff
@@ -0,0 +1,179 @@
+  }))
+  hq.end(`<osm>
+    <changeset>
+      <tag k="comment" v="wow"/>
+    </changeset>
+  </osm>`)
+})
+
+test('add docs to changeset', function (t) {
+  t.plan(13)
+  var href = base + 'changeset/' + changeId + '/upload'
+  var hq = hyperquest.post(href, {
+    headers: { 'content-type': 'text/xml' }
+  })
+  hq.on('response', function (res) {
+    t.equal(res.statusCode, 200)
+    var contentObj = contentType.parse(res)
+    t.equal(contentObj.type, 'text/xml', 'media type correct')
+    t.equal(contentObj.parameters.charset.toLowerCase(), 'utf-8', 'charset correct')
+  })
+  var oldv, newv
+  hq.pipe(concat({ encoding: 'string' }, function (body) {
+    var xml = parsexml(body)
+    t.equal(xml.root.name, 'diffResult')
+    t.deepEqual(xml.root.children.map(function (c) {
+      return c.attributes.old_id
+    }).sort(), [id])
+
+    oldv = version
+    newv = xml.root.children[0].attributes.new_version
+    hyperquest.get(base + 'node/' + version + '/' + newv)
+      .on('response', function (res) {
+        t.equal(res.statusCode, 200)
+      })
+      .pipe(concat({ encoding: 'string' }, onnew))
+    hyperquest.get(base + 'node/' + version + '/' + oldv)
+      .on('response', function (res) {
+        t.equal(res.statusCode, 200)
+      })
+      .pipe(concat({ encoding: 'string' }, onold))
+  }))
+  hq.end(`<osmChange version="1.0" generator="acme osm editor">
+    <modify>
+      <node version="${version}" id="${id}" changeset="${changeId}" lat="111" lon="222"/>
+    </modify>
+  </osmChange>`)
+
+  function onnew (body) {
+    var xml = parsexml(body)
+    t.equal(xml.root.name, 'osm')
+    t.true(isISODate(xml.root.children[0].attributes.timestamp))
+    delete xml.root.children[0].attributes.timestamp
+    t.deepEqual(xml.root.children, [
+      {
+        name: 'node',
+        attributes: {
+          lat: '111',
+          lon: '222',
+          id: id,
+          version: newv,
+          changeset: changeId
+        },
+        children: []
+      }
+    ])
+  }
+  function onold (body) {
+    var xml = parsexml(body)
+    t.equal(xml.root.name, 'osm')
+    t.true(isISODate(xml.root.children[0].attributes.timestamp))
+    delete xml.root.children[0].attributes.timestamp
+    t.deepEqual(xml.root.children, [
+      {
+        name: 'node',
+        attributes: {
+          lat: obs.lat.toString(),
+          lon: obs.lon.toString(),
+          id: id,
+          version: oldv
+        },
+        children: [
+          {
+            name: 'tag',
+            attributes: {
+              k: 'obs',
+              v: 'bar'
+            },
+            children: []
+          }
+        ],
+        content: ''
+      }
+    ])
+  }
+})
+
+test('observation intact when queried', function (t) {
+  osm.get(id, function (err, docs) {
+    t.error(err)
+    var element = docs[0]
+    schema.validateObservation(element)
+    t.deepEquals(obs.attachments, element.attachments)
```

Comment:
Correct version too?

Reply:
I think that is tested in an earlier test (onold, onnew) but can add here if you think it's a good idea. This test as written is for testing the behavior that these extra 'observation-only' fields stayed there